core/include/detray/propagator/actors/pointwise_material_interactor.hpp

```diff
@@ -78,13 +78,18 @@ struct pointwise_material_interactor : actor {
                 const auto mat = detail::material_accessor::get(
                     material_group, mat_index, bound_params.bound_local());
 
-                // return early in case of zero thickness
+                // Return early in case of zero thickness or neutral particle
                 if (mat.thickness() <=
                     std::numeric_limits<scalar_type>::epsilon()) {
                     return false;
                 }
 
                 const scalar_type qop = bound_params.qop();
+                // Only charged particles
+                if (math::fabs(qop) <=
+                    std::numeric_limits<scalar_type>::epsilon()) {
```

Comment:
Why do you want to have this? This assertion won't bring any advantage and `pointwise_material_interactor` is not  only for charged particles. It is true that currently implemented interaction models are only for charged particles but we could add ones for neutral particles later.

Reply:
Dividing by qop leads to errors, for example, when the relativistic quantities are initialized. Do you want me to solve it differently?

Reply:
I believe dividing by qop only happens in charged particles if I am not wrong. We can just add an if-block at https://github.com/acts-project/detray/blob/03c8f8e6cd2df242aa323e6f091c87aa4a9a6eb8/core/include/detray/materials/interaction.hpp#L45 , so it won't call the bethe bloch function on neutral particles

Reply:
I think it happens in the constructor of the relativistic quantities https://github.com/acts-project/detray/blob/03c8f8e6cd2df242aa323e6f091c87aa4a9a6eb8/core/include/detray/materials/detail/relativistic_quantities.hpp#L65 (or am I seeing this wrong?), which is required by all of the public methods of the interaction class

Reply:
There is an assertion on assert(qOverp != 0.f) right above so it looks OK to me

Reply:
Yes, indeed, putting the `pointwise_material_interactor` stops at this assertion if it is compiled into an actor chain that is also used for straight line tracks. Which is my problem

Reply:
Hm, wait, the line above also does not add up, or am I wrong? It is dividing by q if q is zero... So q=0 should also get an assertion

Reply:
relstivistic quantity may be f ixed to be qop-free in the future (so having q and p separately). The if block added in this PR is not just at the right place unless you fix the file name to `charged_particle_interactor`

Reply:
fixed the file name

Reply:
That's not what I meant...

Reply:
Without the if statement, the material validation does not run for me anymore (admittedly I am compiling with a fringe case toolchain). Can we keep the if statement as a hotfix until the relativistic quantities are fixed, please?

Reply:
Can you just leave this part as it is unless you are having an urgent problem caused by this?

Reply:
Just take out the pointwise_material_interactor from your actors on neutral particles

Reply:
I am sorry, but this happens in a general tool which is running correctly and I don't really want to rewrite it because of this. I can only check the charge at runtime, but the actor chain is compile time and part of the propagator type, so that might not be so easy. It is the material interactor that has the bug, if it is meant to be running for charge zero particles as you say.
I will try to run it on another machine. Up until now no other toolchain got stuck on this

Reply:
I left the change in the relativistic quantities, so that it does not divide by `q=0` or is that wrong?

Reply:
if charge is 0, just pick up the second propagator without material interaction. I don't see any reason that is going to be not available

Reply:
I did not say it is impossible, I am saying it is more complicated and I have to change my tools to do this and then back once the material interactor actually does something for uncharged particles. Imo it is better to put a hotfix into the code that has the problem, than its clients.

Reply:
Sure. Just make a better hotfix that I can accept